rqt_py_common/src/rqt_py_common/message_field_type_helpers.py

```diff
@@ -0,0 +1,505 @@
+# Copyright (c) 2020, Michael Lautman PickNik Robotics
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#   * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#   * Redistributions in binary form must reproduce the above
+#     copyright notice, this list of conditions and the following
+#     disclaimer in the documentation and/or other materials provided
+#     with the distribution.
+#   * Neither the name of the TU Darmstadt nor the names of its
+#     contributors may be used to endorse or promote products derived
+#     from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+# Author: Michael Lautman
+# This functionality was ported from ROS1. For the original soure code:
+#   https://github.com/ros/ros_comm/blob/noetic-devel/tools/rostopic/src/rostopic/__init__.py
+
+import re
+
+from typing import Dict, TypeVar, Tuple, Any, Optional
+
+from rclpy import logging
+
+from rqt_py_common.message_helpers import get_message_class
+from python_qt_binding.QtCore import qWarning
+
+from rosidl_parser.definition import BASIC_TYPES, INTEGER_TYPES, \
+                                     FLOATING_POINT_TYPES, CHARACTER_TYPES, \
+                                     BOOLEAN_TYPE, OCTET_TYPE
+
+MsgType = TypeVar('MsgType')
+
+BASIC_STRING_TYPES = (
+    'string',
+    'wstring'
+)
+
+PRIMITIVE_TYPES = (
+    *BASIC_TYPES,
+    *BASIC_STRING_TYPES
+)
+
+STRING_TYPES = (
+    *BASIC_STRING_TYPES,
+    *CHARACTER_TYPES
+)
+
+SEQUENCE_PREFIX = 'sequence<'
+SEQUENCE_PREFIX_LEN = len(SEQUENCE_PREFIX)
+
+__LOGGER = logging.get_logger('message_field_type_helpers')
+
+# leters, integers, and underscore  a slash then more leters, integers, and underscore
+GENERATED_MSG_RE = '[a-zA-Z0-9_]+/[a-zA-Z0-9_]+'
```

Comment:
This can be simplified and I believe forward slashes should generally be escaped.
```suggestion
GENERATED_MSG_RE = '\w+\/\w+'
```

Reply:
I no longer use this. Although that is really good to know for the future.